wxImage::Paste is crashing when target and source images do not have alpha channels

hwiesmann opened the pull request:
Commit 1f0ade29 introduced a bug: in case of not equal image sizes (source and target) pixels have to be copied one by one. This copy procedure does not check if the target has an alpha channel but assumes that it always has an alpha channel (which is not true).

Commit:
Fix crash in wxImage::Paste() when using images without alpha

Check for the presence of alpha channel before using it in the fallback
code.

Also add a unit test exercising this branch of the code and which
crashed before.

vadz commented:
Oops, thanks for noticing (and fixing!) this, will merge soon. I just wonder how this hasn't been caught by the tests...

vadz committed:
Fix running new wxImage::Paste() unit tests from build directory

Add new image test files added in 6e8da8641c (Add alpha blending for
wxImage::Paste, 2020-09-23) and 1f0ade29f0 (Fix using mask colour even
if there is no mask in wxImage::Paste, 2020-09-30) to Makefile, so that
they're copied to the build directory and could be found by the test
there, otherwise running wxImage::Paste test when building outside of
the source directory failed.

vadz committed:
Rearrange sections in wxImage::Paste unit test

No real changes, just execute only the tests relevant for the section
being executed instead of always running the code loading PNG images
when running a section not using them at all.

This has the advantages of being a bit faster to execute (and the total
test suite run time does add up) and, more importantly, showing the
right amount of tests when executing a single section: e.g. now it's 1
in the simplest case, as expected, and not 5.

The disadvantage is, of course, slightly more code duplication, but it
doesn't seem a huge problem in practice and making each section code
more self-contained is arguably worth it.

Also, compensate for this somewhat by extracting all XPM definitions at
the function level: those are only initialized once (being static), so
there is no need to duplicate them.

Finally, also call AddHandler() once instead of once per section as
before.

vadz commented:
Sorry for force-pushing into your branch, but I wanted to add the new test before merging this and had to make a couple more changes before being able to add it. I'll merge it once the CI builds pass.

Thanks again!
